flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStResourceContainer.java

```diff
@@ -436,7 +441,7 @@ public void clearDirectories() throws Exception {
     }
 
     public void forceClearRemoteDirectories() throws Exception {
-        if (remoteBasePath != null) {
+        if (remoteBasePath != null && remotePathNewlyCreated) {
```

Comment:
I am unsure what this is doing. I would think for a method called force clear we should clear the contents, whether remotePathNewlyCreated is true or false. Could content be created in the directory when remotePathNewlyCreated = false?